Inputs in dedicated Task

bmorcelli opened the pull request:
#### Proposed Changes ####

* Changed the functions checkPrevPress, checkSelPress, checkNextPress, checkEscPress and checkAnyKeyPress to volatile bool variables and made them run on a InputHandler function on a dedicated task.

* Added debounce into the InputHandler, removing some delays from the code (the ones before and after loopOption callings.

* Touch bar for CYD, Core2 and CoreS3 is now the whole screen (left side for prev, center for Select and right for Next, more options can be added now, if needed, like Esc btn as an [X] on screen)

#### Types of Changes ####

Structural 

#### Verification ####

Overall speed might be improved in loopTask.

#### Testing ####

Need to test on all devices, can be tested in [this flasher](https://bmorcelli.github.io/Bruce/flasher.html) (using BetaRelease)

# DO NOT MERGE YET, Please...

@rouing Can you have a look on it?

bmorcelli committed:
Inputs running on task, removed delays

Inputs (chechSelPress and stuff) are checked in a specific task, running on background

Removed delay before and after loopOptions

bmorcelli committed:
oops

rouing reviewed (approved):
Nice job
Although needs some adjustments, I still give my thumbs up.

bmorcelli committed:
Task control and var handling
Added a function to control the Input variables, now they "rearm" after read.

Fixed Restarts and some other changes

Fixed Keyboard to send Delete key and modifiers https://github.com/pr3y/Bruce/issues/594

bmorcelli committed:
Interface Updates
Removed keyboard function from interface.cpp, all devices are now usinf the same function, easy to maintain

Remapped TouchArea for touchscreen devices.. now it is possible to use the touch area to escape some functions, go up/down/left/right and selec

bmorcelli committed:
Building errors

bmorcelli committed:
goto errors

bmorcelli committed:
Boot tft speedup

bmorcelli committed:
checkLetterShortcutPress glitch

bmorcelli committed:
CYD crashing

bmorcelli committed:
Fix CYD start and Core2/S3 debounce

bmorcelli committed:
RF Scan/Copy "Main Menu" option fix

bmorcelli committed:
M5Unified and M5GFX removal
These libs are now being included in the specific environment, getting t

bmorcelli committed:
.
